src/impls/mod.rs

```diff
@@ -12,3 +12,8 @@ mod serde_json;
 
 #[cfg(feature = "ijson")]
 mod ijson;
+
+#[cfg(feature = "indexmap")]
+mod indexmap;
+#[cfg(feature = "indexmap")]
+mod indexset;
```

Comment:
I would really prefer to have each external crate implemented in a single separate module, just like you did at first. You can solve the clippy warning either by renaming the submodules in `indexmap` `map` and `set`, or by adding `#[allow(clippy::module_inception)]` (I think it can be a justified use of module inception).